addons/mail/static/tests/discuss_app/inbox.test.js

```diff
@@ -623,3 +627,43 @@ test("emptying inbox doesn't display rainbow man in another thread", async () =>
     // weak test, no guarantee that we waited long enough for the potential rainbow man to show
     await contains(".o_reward_rainbow", { count: 0 });
 });
+
+test("Counter should update properly when a new message is added to inbox", async () => {
```

Comment:
update properly = be incremented by 1 (better be explicit)

it's not just "a new message is added to inbox" but more specifically "when receiving a message with a mention in a channel" (and inbox is enabled for the target user, which we're currently not checking in the test)